src/lib/tasks/server/send_query_response_task.cpp

```diff
@@ -63,13 +67,13 @@ void SendQueryResponseTask::_send_row_description() {
         Fail("Bad DataType");
     }
 
-    PostgresWireHandler::write_value(output_packet, htonl(object_id));   // object id of type
-    PostgresWireHandler::write_value(output_packet, htons(type_width));  // regular int
-    PostgresWireHandler::write_value(output_packet, htonl(-1));          // no modifier
-    PostgresWireHandler::write_value(output_packet, htons(0u));          // text format
+    PostgresWireHandler::write_value(*output_packet, htonl(object_id));   // object id of type
+    PostgresWireHandler::write_value(*output_packet, htons(type_width));  // regular int
+    PostgresWireHandler::write_value(*output_packet, htonl(-1));          // no modifier
+    PostgresWireHandler::write_value(*output_packet, htons(0u));          // text format
   }
 
-  _session->async_send_packet(output_packet);
+  _connection->_send_bytes_async(output_packet, false);
```

Comment:
the naming of send_bytes_async is wrong, i.e. _ is for private methods

Reply:
this method ist not public either. what about protected methods? are they with or without underscore?

Reply:
protected == private in hyrise. There are very few real private methods, most of them are protected

Reply:
The real issue here is, that SendQueryResponseTask is a friend class of ClientConnection -- this is a purposefully ugly way of making this work.

The SendQueryResponseTask is the only place where our layering is broken -- and I'm still not entirely convinced that just for the purpose of string concatenation on the scheduler thread, we should open ourselves up to such architectural issues and possibly threading problems (now both the Task and the HyriseSession could potentially manipulate the ClientConnection in parallel).

Ideally, we'd pass the ClientConnection into the HyriseSession as a unique_ptr, then we'd have the following layers:

Server Tasks
----
HyriseSession
----
ClientConnection + PostgresWireHandler
----
Boost Asio Socket